bonfire/namespaces.py

```diff
@@ -264,6 +264,8 @@ def reserve_namespace(name, requester, duration, timeout, local=True):
         requester,
         duration,
     )
+    if not ns_name:
+        raise FatalError("Reservation of namespace failed.")
```

Comment:
@Jason-RH is this good w/ you?

Reply:
Yeah this should be fine, but @psegedy just fyi the reservation may still get reconciled eventually. We should change it so that bonfire deletes the reservation on timeout.. but we can resolve that separately

Reply:
Will address reservation deletion in a follow-up